weblink/_internal/Server.hx

```diff
@@ -1,32 +1,58 @@
 package weblink._internal;
 
-import haxe.MainLoop;
+import haxe.EntryPoint;
+import haxe.Exception;
+import haxe.Timer;
 import haxe.io.Bytes;
-import hl.uv.Stream;
+import hl.Gc;
+import hl.uv.Loop;
 import sys.net.Host;
+import sys.thread.EventLoop;
+import sys.thread.Lock;
+import sys.thread.Thread;
 import weblink._internal.Socket;
 
 class Server extends SocketServer {
+	/**
+		Is the server currently running?
+	**/
+	public var running:Bool;
 
-	var parent:Weblink;
-	public var running:Bool = true;
-	var loop:hl.uv.Loop;
-
-	public function new(port:Int, parent:Weblink) {
-		// sockets = [];
-		loop = hl.uv.Loop.getDefault();
-		super(loop);
-		bind(new Host("0.0.0.0"), port);
-		noDelay(true);
-		listen(100, function() {
-			final stream = accept();
-			final socket:Socket = cast stream;
+	private final parent:Weblink;
+	private final uvLoop:Loop;
+	private var serverThread:Null<Thread>;
+	private var helperTimer:Null<Timer>;
+
+	public function new(app:Weblink) {
+		this.uvLoop = @:privateAccess Loop.default_loop(); // don't register MainLoop event
+		super(this.uvLoop);
+
+		this.parent = app;
+		this.running = false;
+	}
+
+	public function start(host:Host, port:Int, model:StartModel) {
+		final lock = new Lock();
+
+		// Prepare the libuv TCP socket
+		super.bind(host, port);
+		super.noDelay(true);
+
+		// Configure new connection callback
+		super.listen(100, function() {
+			Gc.enable(true); // Allow to stop the world when we're not polling
+
+			final client = this.accept();
+
+			// Register a handler for incoming data (HTTP/1.1 specific)
 			var request:Null<Request> = null;
+			client.readStart(function(data:Null<Bytes>) @:privateAccess {
+				Gc.enable(true);
 
-			stream.readStart(function(data:Null<Bytes>) @:privateAccess {
 				if (data == null) { // EOF
 					request = null;
-					stream.close();
+					client.close();
+					Gc.enable(false);
```

Comment:
I can't comprehend why after an EOF disabling the GC is a good idea, it will simply be reenabled when another client sends data to the server.